plugin/router/json_router.go

```diff
@@ -78,20 +78,18 @@ type JSONErrorHandler func(err plugin.Error) (int, JSONResponse)
 type JSONRouter struct {
 	*Subrouter
 	errHandler JSONErrorHandler
-	logger     log.Logger
 }
 
 // NewJSONRouter returns a new JSONRouter with a logger.
-func NewJSONRouter(logger log.Logger) *JSONRouter {
-	return NewJSONRouterWithErrorHandler(logger, nil)
+func NewJSONRouter() *JSONRouter {
```

Comment:
I don't think we have to solve this problem immediately in this PR but I'd like to point out a problem of knowing where in code a log line has come from. Previously I've been hand-rolling a combination of a required "component" field and merging logging attributes between a default component logger and a logger from context – I wonder if we could automate this somehow. One option would be to use something like `runtime.Caller`, another to keep passing loggers to initialisers and attaching a `component` field to them.

Reply:
True, I also wanted to make sure that if a Logger is set upstream in the request context, the one used for error logging by the router is the one set in the context, rather than a different one. Probably good to add some additional standard fields that can be attached with `With` in both the router calls and in upstream logging.

Reply:
Makes sense to me – I think we can address that in a follow-up issue / PR 👍